src/core/coreconfig.cpp

```diff
@@ -40,6 +41,9 @@ void CoreConfig::init(const QJsonObject &p_app,
     if (m_toolBarIconSize <= 0) {
         m_toolBarIconSize = 16;
     }
+
+    if(!isUnDefinedKey(appObj, userObj, "minimize_to_system_tray"))
+        m_minimize_to_system_tray = int(READBOOL(QStringLiteral("minimize_to_system_tray")));
```

Comment:
How about just use `int` to represent this value instead of `bool`?

- `-1` for prompting for user;
- `0` for disabling minimizing to system tray;
- `1` for enabling minimizing to system tray;

Reply:
I assume there are 3 states in the user profile:
Undefined (there is no such key)
true
false

In init, the default value is -1. If the user defines, convert bool to int.